src/schemes.hpp

```diff
@@ -0,0 +1,293 @@
+// Copyright 2020 Chia Network Inc
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in coiance with the License.
+// You may obtain a copy of the License at
+
+//    http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or iied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SRC_BLSSCHEMES_HPP_
+#define SRC_BLSSCHEMES_HPP_
+
+#include <iostream>
+#include <vector>
+
+#include "relic_conf.h"
+
+#if defined GMP && ARITH == GMP
+#include <gmp.h>
+#endif
+
+#include "elements.hpp"
+#include "privatekey.hpp"
+#include "util.hpp"
+// #include "publickey.hpp"
+
+using std::vector;
+
+// These are all MPL schemes
+namespace bls {
+
+class Core {
+    friend class BasicScheme;
+    friend class AugScheme;
+    friend class PopScheme;
+
+public:
+    // static PrivateKey KeyGen(... ikm);  TODO
+    static vector<uint8_t> SkToPk(const PrivateKey &seckey);
+
+    static G1Element SkToG1(const PrivateKey &seckey);
+
+    static vector<uint8_t> Sign(
```

Comment:
I think we should remove this, we can always call G2Element.serialize() right? What's the point of adding the extra method?